packages/runtime/src/next/components/page.tsx

```diff
@@ -1,38 +1,21 @@
 'use client'
 
-import { Suspense, memo, useMemo } from 'react'
+import { memo } from 'react'
 
-import { useDispatch } from '../../runtimes/react/hooks/use-dispatch'
-import { useUniversalDispatch } from '../../runtimes/react/hooks/use-universal-dispatch'
-import { useCacheData } from '../../runtimes/react/hooks/use-cache-data'
+import { type MakeswiftPageSnapshot } from '../../client'
 
-import { Page as PageComponent } from '../../components/page'
 import {
-  type MakeswiftPageSnapshot,
-  type MakeswiftPageDocument,
-  pageToRootDocument,
-} from '../../client'
-import * as ReactPage from '../../state/react-page'
-import { registerDocumentsEffect } from '../../state/actions'
-import { type PageMetadataSettings } from '../../components/page/page-seo-settings'
+  type PageMetadataSettings,
+  Page as BuiltinPage,
+} from '../../runtimes/react/components/page'
+
+import { useRouterLocaleSync } from '../hooks/use-router-locale-sync'
 
 export type PageProps = {
   snapshot: MakeswiftPageSnapshot
   metadata?: boolean | PageMetadataSettings
 }
 
-function useRegisterPageDocument(pageDocument: MakeswiftPageDocument): ReactPage.Document {
```

Comment:
Moved to framework-independent implementation here: https://github.com/makeswift/makeswift/pull/1071/files#diff-13bd7ab9a65a55cc9147b0bf5a16cd5530d82f9a07c522486cf4444185ed0d99R21